co-simulation/fed/mosaic-carma-cloud/src/main/java/org/eclipse/mosaic/fed/carmacloud/ambassador/CarmaCloudMessageAmbassador.java

```diff
@@ -0,0 +1,224 @@
+		carmaCloudRegistrationReceiver.init();
+		registrationRxBackgroundThread = new Thread(carmaCloudRegistrationReceiver);
+		registrationRxBackgroundThread.start();
+
+		try
+		{
+			rti.requestAdvanceTime(currentSimulationTime, 0, (byte) 1);
+		}
+		catch (IllegalValueException e)
+		{
+			log.error("Error during advanceTime request", e);
+			throw new InternalFederateException(e);
+		}
+	}
+
+
+	/**
+	 * This method processes the interactions.
+	 *
+	 * @param interaction The interaction that can be processed.
+	 * @throws InternalFederateException Exception is thrown if an error is occurred
+	 *                                   while execute of a federate.
+	 */
+	@Override
+	public void processInteraction(Interaction interaction)
+		throws InternalFederateException
+	{
+		String type = interaction.getTypeId();
+		long interactionTime = interaction.getTime();
+		log.trace("Process interaction with type '{}' at time: {}", type, interactionTime);
+	}
+
+
+	/**
+	 * This method is called by the AbstractFederateAmbassador when the RTI grants a
+	 * time advance to the federate.Any unprocessed interactions are forwarded to
+ the federate using the processInteraction method before this call is made.
+	 *
+	 * @param time The timestamp (in nanoseconds) indicating the time to which the federate can
+	 *             advance its local time.
+	 * @throws InternalFederateException
+	 */
+	@Override
+	public synchronized void processTimeAdvanceGrant(long time)
+		throws InternalFederateException
+	{
+		// Process the time advance only if the time is equal or greater than the next
+		// simulation time step
+		log.info("processTimeAdvanceGrant {} {}", currentSimulationTime, time);
+		if (time < currentSimulationTime)
+				return;
+
+		log.info("CarmaCloud message ambassador processing timestep to {}.", time);
+		try
+		{
+			// Handle any new carmaCloud registration requests
+			List<CarmaCloudRegistrationMessage> newRegistrations = carmaCloudRegistrationReceiver.getReceivedMessages();
+			for (CarmaCloudRegistrationMessage reg : newRegistrations)
+			{
+				log.info("Processing new registration request for  {}.", reg.getId());
+				// Store new instance registration to carmaCloud instance manager
+				carmaCloudInstanceManager.onNewRegistration(reg);
+			}
+
+			timeSyncSeq += 1;
+			// nanoseconds to milliseconds for CarmaCloudTimeMessage
+			TimeSyncMessage timeSyncMessage = new TimeSyncMessage(currentSimulationTime / 1000000L, timeSyncSeq);
+			carmaCloudInstanceManager.onTimeStepUpdate(timeSyncMessage);
+
+			// Advance the simulation time
+			currentSimulationTime += carmaCloudConfiguration.updateInterval * TIME.MILLI_SECOND;
+
+			// Request the next time advance from the RTI
+			log.info("Requesting timestep updated to  {}.", currentSimulationTime);
+			rti.requestAdvanceTime(currentSimulationTime, 0, (byte) 2);
+		}
+		catch (IllegalValueException e)
+		{
+				throw new InternalFederateException(e);
+		}
+		catch (IOException e1)
```

Comment:
Nit: Since the previous catch statement and this one do not share scope we can call both exceptions `e`.  Not high priority but adding weird unnecessary naming conventions like this makes for harder to read code.